nsuns-client/src/hooks/queries/auth.ts

```diff
@@ -0,0 +1,51 @@
+import { createMutation, createQuery, useQueryClient } from "@tanstack/solid-query";
+import { QueryKeys } from "./keys";
+import { login, loginAnonymous, logout, userInfo } from "../../api/auth";
+
+export const useUserInfoQuery = () => {
+  return createQuery({
+    queryKey: QueryKeys.auth,
+    queryFn: userInfo,
+    cacheTime: Infinity,
+    staleTime: Infinity
+  });
+};
+
+export const useLogoutMutation = () => {
+  const queryClient = useQueryClient();
+  const mutation = createMutation({
+    mutationFn: logout,
+    onSuccess: () => {
+      queryClient.clear();
+      window.location.replace("/login");
```

Comment:
Doing window.location.replace so the browser will hard-refresh, and clear any cached queries.